python/lsst/butler/quantum.py

```diff
@@ -0,0 +1,85 @@
+# 
+# LSST Data Management System
+#
+# Copyright 2008-2017  AURA/LSST.
+# 
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# 
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+# 
+# You should have received a copy of the LSST License Statement and 
+# the GNU General Public License along with this program.  If not, 
+# see <https://www.lsstcorp.org/LegalNotices/>.
+#
+
+from .datasets import DatasetRef
+from .run import Run
+
+
+class Quantum:
+
+    __slots__ = ("_quantumId", "_registryId", "_run", "_task", "_predictedInputs", "_actualInputs")
+
+    _currentId = -1
```

Comment:
Both are pre-increment, so should start at `-1`. Note that the whole ID thing is ugly, flaky and needs to be fixed. The only reason for it existing is because `sqlite3` doesn't allow `autoincrement` compound-primary-keys. I should probably create a ticket for this...

Reply:
DM-13063 created.